src/v/cloud_storage/partition_manifest.cc

```diff
@@ -262,6 +262,16 @@ const model::offset partition_manifest::get_last_offset() const {
     return _last_offset;
 }
 
+const std::optional<kafka::offset>
+partition_manifest::get_last_kafka_offset() const {
+    auto last_seg = last_segment();
+    if (!last_seg.has_value()) {
+        return std::nullopt;
+    }
+    // TODO: handle when delta_offset_end isn't in the segment.
```

Comment:
This situation is handled by the `segment_meta::committed_kafka_offset` method. It will return redpanda offset. But this shouldn't be a problem unless someone is using old redpanda.

Reply:
Yeah, I left this TODO in case of dealing with old manifests. In those cases, the only solutions I've come up with are all really heavyweight